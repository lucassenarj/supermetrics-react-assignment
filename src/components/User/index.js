import React from "react";
import PropTypes from "prop-types";
import "./index.scss";

function User({ id, name, onClick, posts }) {
  return (
    <button className="user" onClick={onClick} data-testid={id}>
      <p>{ name } </p> 
      <p className="user__count">{ posts }</p>
    </button>
  );
};

User.propTypes = {
  name: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  posts: PropTypes.number.isRequired,
};

export default User;
