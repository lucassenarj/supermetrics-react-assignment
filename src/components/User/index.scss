@import "./../../assets/colors.scss";

.user {
  background-color: $backgroundUser;
  padding: 5px 20px;
  margin: 20px auto;
  cursor: pointer;
  border: solid 2px #a5a16f;
  border-radius: 10px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 100%;
}

.user__count {
  background-color: $backgroundUserCount;
  padding: 20px;
  border-radius: 50px;
  width: 20px;
  text-align: center;
}