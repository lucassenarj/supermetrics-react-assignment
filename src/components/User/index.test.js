import React from "react";
import { render } from "@testing-library/react";
import User from "./";

const values = {
  name: "User test",
  onClick: jest.fn(),
  posts: 3,
}

describe("Components", () => {
  describe("User", () => {
    it("renders correctly", () => {
      const { container } = render(<User
        name={values.name}
        onClick={values.onClick}
        posts={values.posts}
      />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
