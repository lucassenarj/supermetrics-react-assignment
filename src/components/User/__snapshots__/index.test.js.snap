// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components User renders correctly 1`] = `
<button
  class="user"
>
  <p>
    User test
     
  </p>
  <p
    class="user__count"
  >
    3
  </p>
</button>
`;
